radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/epoch/state/PreparingUnstake.java

```diff
@@ -0,0 +1,143 @@
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.application.system.scrypt.epoch.state;
+
+import com.radixdlt.application.system.scrypt.EpochUpdateConfig;
+import com.radixdlt.application.system.scrypt.EpochUpdateConstraintScrypt;
+import com.radixdlt.application.system.scrypt.ValidatorScratchPad;
+import com.radixdlt.application.tokens.state.PreparedUnstakeOwnership;
+import com.radixdlt.constraintmachine.IndexedSubstateIterator;
+import com.radixdlt.constraintmachine.ReducerState;
+import com.radixdlt.constraintmachine.exceptions.ProcedureException;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.utils.KeyComparator;
+import com.radixdlt.utils.UInt256;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+public final class PreparingUnstake implements ReducerState {
+  private final UpdatingEpoch updatingEpoch;
+  private final NavigableMap<ECPublicKey, NavigableMap<REAddr, UInt256>> preparingUnstakes =
+      new TreeMap<>(KeyComparator.instance());
+  private final NavigableMap<ECPublicKey, NavigableMap<REAddr, UInt256>> preparingStake;
+  private final NavigableMap<ECPublicKey, ValidatorScratchPad> updatingValidators;
+  private final EpochUpdateConfig config;
+
+  private PreparingUnstake(
+      EpochUpdateConfig config,
+      UpdatingEpoch updatingEpoch,
+      NavigableMap<ECPublicKey, ValidatorScratchPad> updatingValidators,
+      NavigableMap<ECPublicKey, NavigableMap<REAddr, UInt256>> preparingStake) {
+    this.config = config;
+    this.updatingEpoch = updatingEpoch;
+    this.updatingValidators = updatingValidators;
+    this.preparingStake = preparingStake;
+  }
+
+  static PreparingUnstake create(
```

Comment:
Why do we have this and not just use the constructor out of interest?

Reply:
Unfinished refactoring. Fixed.